sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -1052,6 +1052,13 @@ object RapidsConf {
     .booleanConf
     .createWithDefault(true)
 
+  val REGEXP_MAX_STATE_MEMORY = conf("spark.rapids.sql.regexp.maxStateMemory")
+    .doc("Specifies the maximum memory on GPU to be used for regular expressions." +
+      "Note that the memory usage is an estimate based on a worst case using only " +
+      "the regular expression pattern.")
+    .longConf
+    .createWithDefault(0x01FFFFFFFFL)
```

Comment:
8-ish GiB is too big. We tell users that we are going to try and keep GPU memory usage under 4x the target batch size is per task, but that includes the batch that the input is a part of.  I would prefer to have us set this closer to what the batch size is and no more than 3x the input batch size.  Could we have the default be 1x or 2x the input batch size and have some smarts when reading it that will warn the user if they set it to more than 3x the input batch size?